### ISSUE TITLE
Parameterized view cannot parse DateTime64 http parameter

### ISSUE THREAD

**Issue by BenoitRanque** [ClickHouse/ClickHouse#69058]
Tested on 24.1 (cannot upgrade due to #65402)

**Company**
Hasura

**Problem**
`DateTime64` HTTP parameters are not parsed correctly when passed to parameterized views.

Simple repro:
```sql
CREATE TABLE events (event_id Int32, timestamp DateTime64) ENGINE=MergeTree();

INSERT INTO events (event_id, timestamp) VALUES
(1, '2019-01-01 00:00:00'),
(2, '2019-01-02 00:00:00'),
(3, '2019-01-03 00:00:00'),
(4, '2019-01-04 00:00:00'),
(5, '2019-01-05 00:00:00');

CREATE VIEW events_view AS
SELECT * FROM events
WHERE timestamp >= {from_date: DateTime64}
AND timestamp <= {to_date: DateTime64};

-- this works as expected:
SELECT * FROM events_view(from_date='2019-01-02 00:00:00', to_date='2019-01-04 00:00:00');
```
The same query, but parameterized:
```http
POST http://localhost:8123?param_from_date=2019-01-02 00:00:00&param_to_date=2019-01-04 00:00:00
X-ClickHouse-User: default
X-ClickHouse-Key: default

SELECT * FROM events_view(from_date={from_date: DateTime64}, to_date={to_date: DateTime64});
```
Fails with:
```
Code: 41. DB::Exception: Cannot parse DateTime: value '1546387200' cannot be parsed as DateTime64 for query parameter 'from_date'. (CANNOT_PARSE_DATETIME) (version 24.1.8.22 (official build))
```

**Comment by BenoitRanque** [comment]
Update: the following works. Perhaps this is not a bug?
```http
POST http://localhost:8123?param_from_date=2019-01-02 00:00:00&param_to_date=2019-01-04 00:00:00
X-ClickHouse-User: default
X-ClickHouse-Key: default

SELECT * FROM events_range(from_date={from_date: String}, to_date={to_date: String})
```

**Comment by BenoitRanque** [comment]
Seems to be fixed in #67654

**Comment by BenoitRanque** [comment]
Confirming this is fixed in `head`

The workaround described above (use a String as the parameter type in the top level) works in 24.1,
and it still works in `head` but only for simple types.
For complex types, we get an error which makes some sense I guess?

Anyways, closing as this bug is fixed on `head`
